plugins/meme.lua

```diff
@@ -0,0 +1,199 @@
+   for k,v in pairs(list1) do
+      for k2, v2 in pairs(list2) do
+         if v2:find(v) then
+            n = n + 1
+         end
+      end
+   end
+   return n
+end
+
+local function match_meme(name)
+   local _memes = load_data()
+   local name = name:lower():split(' ')
+   local max = 0
+   local id = nil
+   for k,v in pairs(_memes.memes) do
+      local n = match_n_word(name, v:lower():split(' '))
+      if n > 0 and n > max then
+         max = n
+         id = v
+      end
+   end
+   return id
+end
+
+local function generate_meme(id, textup, textdown)
+   local base = "http://apimeme.com/meme"
+   local arguments = {
+      meme=id,
+      top=textup,
+      bottom=textdown
+   }
+   return base .. "?" .. helpers.url_encode_arguments(arguments)
+end
+
+local function get_all_memes_names()
+   local _memes = load_data()
+   local text = 'Last time: ' .. _memes.last_time .. '\n-----------\n'
+   for k, v in pairs(_memes.memes) do
+      text = text .. '- ' .. v .. '\n'
+   end
+   text = text .. '--------------\n' .. 'You can see the images here: http://apimeme.com/'
+   return text
+end
+
+local function callback_send(cb_extra, success, data)
+   if success == 0 then
+      send_msg(cb_extra.receiver, "Something wrong happened, probably that meme had been removed from server: " .. cb_extra.url, ok_cb, false)
+   end
+end
+
+local function run(msg, matches)
+   local receiver = get_receiver(msg)
+   if matches[1] == 'list' then
+      local _memes = load_data()
+      return 'I have ' .. #_memes.memes .. ' meme names.\nCheck this link to see all :)\n' .. _memes.link
+   elseif matches[1] == 'listall' then
+      if not is_sudo(msg) then
+         return "You can't list this way, use \"!meme list\""
+      else
+         return get_all_memes_names()
+      end
+   elseif matches[1] == "search" then
+      local meme_id = match_meme(matches[2])
+      if meme_id == nil then
+         return "I can't match that search with any meme."
+      end
+      return "With that search your meme is " .. meme_id
+   end
+   local searchterm = string.gsub(matches[1]:lower(), ' ', '')
+
+   local meme_id = _cache[searchterm] or match_meme(matches[1])
+   if not meme_id then
+      return 'I don\'t understand the meme name "' .. matches[1] .. '"'
+   end
+
+   _cache[searchterm] = meme_id
+   print("Generating meme: " .. meme_id .. " with texts " ..  matches[2] .. ' and ' .. matches[3])
+   local url_gen = generate_meme(meme_id, matches[2], matches[3])
+   send_photo_from_url(receiver, url_gen, callback_send, {receiver=receiver, url=url_gen})
+
+   return nil
+end
+
+return {
+   description = "Generate a meme image with up and bottom texts.",
+   usage = {
+      "-----------------------------------------------------------------------------------------------------------------------------",
+      "Meme commands:",
+      "-----------------------------------------------------------------------------------------------------------------------------",
```

Comment:
And yeah, in mobile are 2 lines and a little bit more.
Probably this need to be a little bit less XD